src/lib.rs

```diff
@@ -10,11 +10,34 @@ pub enum Error {
     Checksum,
 }
 
+/// Protected ID which is a 6 bit ID with two parity bits
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct PID(u8);
+
+impl PID {
+    /// Calculate the PID from an ID.
+    /// P0 = ID0 ⊕ ID1 ⊕ ID2 ⊕ ID4
+    /// P1 = ¬(ID1 ⊕ ID3 ⊕ ID4 ⊕ ID5)
+    pub fn from_id(id: u8) -> PID {
+        assert!(id < 64, "ID must be less than 64");
+        // count parity bits and check if they are even odd
+        let p0 = (id & 0b10111).count_ones() as u8 & 0b1;
+        let p1 = ((id & 0b111010).count_ones() as u8 + 1) & 0b1;
+        println!("p0: {}, p1: {}", p0, p1);
```

Comment:
Do you really want `println` in a library?

Reply:
Oh good catch, that was for debugging.

Reply:
Fixed.